### PR TITLE
Automatically generate SAWScript boilerplate from LLVM AST

### PR DESCRIPTION
**Pull request by chameco**
(Depends on GaloisInc/crucible#299.)

This PR introduces three `Experimental` SAWScript functions:

- `crucible_llvm_array_size_profile : LLVMModule -> String -> CrucibleSetup () -> TopLevel [Profile]` symbolically executes a function and returns a list of profiles. These describe the sizes of buffers referenced by pointer arguments to all functions called during symbolic execution.
- `llvm_boilerplate : String -> LLVMModule -> [Profile] -> TopLevel ()` writes SAWScript proof "boilerplate" (method specs, calls to `crucible_llvm_verify` with appropriate overrides, etc.) to the path specified in the first argument. If profiles are available, they are used to guess array sizes.
- `llvm_boilerplate_info : LLVMModule -> [Profile] -> TopLevel ()` prints an internal representation of the boilerplate (for debugging).

### CODE REVIEW

File: src/SAWScript/Crucible/LLVM/Builtins.hs [hunk 1]
```diff
@@ -780,8 +796,14 @@ verifySimulate opts cc mspec args assumes top_loc lemmas globals checkSat =
     invariantExecFeatures <- mapM
       (registerInvariantOverride opts cc top_loc (HashMap.fromList breakpoints))
       (groupOn (view csName) invLemmas)
+
+    additionalFeatures <- case asp of
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/LLVM/Builtins.hs]: maybe use `maybeToList `

File: src/SAWScript/Crucible/LLVM/Boilerplate.hs [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveFoldable #-}
+{-# LANGUAGE DeriveTraversable #-}
+
+module SAWScript.Crucible.LLVM.Boilerplate
+  ( llvm_boilerplate_info
+  , llvm_boilerplate
+  ) where
+
+import System.IO (FilePath, IOMode(..), stderr, withFile)
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/LLVM/Boilerplate.hs]: same comment about `import`

File: src/SAWScript/Crucible/LLVM/Boilerplate.hs [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+import qualified Lang.Crucible.LLVM.TypeContext as Crucible
+import qualified Lang.Crucible.LLVM.MemType as Crucible
+
+import Lang.Crucible.LLVM.ArraySizeProfile (Profile)
+
+newtype BPException = BPException Text
+  deriving Show
+instance Exception BPException
+
+data BPType = BPVoid
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/LLVM/Boilerplate.hs]: any reason not to have `Eq` and `Ord`?

**Comment by langston-barrett** [comment]
This passed the Fryingpan tests: http://fryingpan.dev.galois.com/hydra/jobset/s2n/PR-llvm-boilerplate.submodules-ghc865#tabs-jobs

**Pull request merged**